### PR TITLE
Modify repository to use the correct skeleton repository

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request modifies the repository to use [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) as its Lineage parent.

## 💭 Motivation and context ##

This repo previously had _no_ Lineage parent and was based on [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic).  [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) is a better fit for this repository.

Note that the changes in https://github.com/cisagov/dmarc-import-tf-module/pull/45 were necessary for this Terraform to be applied correctly. 

## 🧪 Testing ##

All automated testing passes.  

Additionally, this Terraform was applied in Production, mostly successfully.  Unfortunately, it could not be completely applied because of the following error:

```
| Error: error modifying Lambda Function (dmarc-import) Code: InvalidParameterValueException: The runtime parameter of python3.6 is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (python3.9) while creating or updating functions.
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "2992b212-9f78-4c00-918b-6cc4fd11e24a"
│   },
│   Message_: "The runtime parameter of python3.6 is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (python3.9) while creating or updating functions.",
│   Type: "User"
│ }
│ 
│   with module.dmarc_import.aws_lambda_function.lambda,
│   on .terraform/modules/dmarc_import/lambda.tf line 2, in resource "aws_lambda_function" "lambda":
│    2: resource "aws_lambda_function" "lambda" {
```

Despite that, @jsf9k and I confirmed that after applying this Terraform in Production, the Lambda continued to function and we were able to see new records flowing into the database.

@jsf9k is going to create an issue and PR to update [`cisagov/dmarc-import-lambda`](https://github.com/cisagov/dmarc-import-lambda) from python 3.6 to a more modern version.  Waiting for that PR should not prevent this PR from moving forward.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
So far, so good!  Please re-request my review once this repo is up and running against Terraform 1.0.

**Comment by dav3r** [comment]
@jsf9k, please take a look at the updated PR description above and make sure you're ok with it.  Since you created this PR initially, you cannot be added as a reviewer, but please take a look and note any concerns you might have.

@mcdonnnj and @jasonodoom - this PR is now ready for your review.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    schedule:
+      interval: "weekly"
+
+  - package-ecosystem: "terraform"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+    # # Managed by cisagov/skeleton-tf-module
+    # ignore:
+    #   - dependency-name: "hashicorp/aws"
```
Comment (jsf9k) [inline .github/dependabot.yml]: Uncomment ignore directive:
```suggestion
    # Managed by cisagov/skeleton-tf-module
    ignore:
      - dependency-name: "hashicorp/aws"
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  - package-ecosystem: "terraform"
+    directory: "/examples/basic_usage"
+    schedule:
+      interval: "weekly"
+    # # Managed by cisagov/skeleton-tf-module
+    # ignore:
+    #   - dependency-name: "hashicorp/aws"
```
Comment (jsf9k) [inline .github/dependabot.yml]: Uncomment ignore directive:
```suggestion
    # Managed by cisagov/skeleton-tf-module
    ignore:
      - dependency-name: "hashicorp/aws"
```

File: provisiondmarcimport_policy.tf [hunk 3]
```diff
@@ -102,7 +103,6 @@ data "aws_iam_policy_document" "provisiondmarcimport" {
       "logs:DeleteResourcePolicy",
       "logs:DeleteRetentionPolicy",
       "logs:ListTagsLogGroup",
-      "logs:PutResourcePolicy",
```
Comment (jsf9k) [inline provisiondmarcimport_policy.tf]: I suspect that `logs:PutResourcePolicy` could come back to live here if we changed `arn:aws:logs:${var.aws_region}:${local.dns_account_id}:log-group:/aws/aes/domains/${var.elasticsearch_domain_name}/application-logs*` to `arn:aws:logs:${var.aws_region}:${local.dns_account_id}:log-group:/aws/aes/domains/${var.elasticsearch_domain_name}/application-logs:*` and `arn:aws:logs:${var.aws_region}:${local.dns_account_id}:log-group:/aws/lambda/${var.lambda_function_name}*` to `arn:aws:logs:${var.aws_region}:${local.dns_account_id}:log-group:/aws/lambda/${var.lambda_function_name}:*` in the `resources` block.

I'm not certain about this, though.

**Review by jsf9k — COMMENTED** [review]
I have a few changes to request, plus one comment.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    ignore:
+      - dependency-name: "hashicorp/aws"
+
+  - package-ecosystem: "terraform"
+    directory: "/examples/basic_usage"
+    schedule:
+      interval: "weekly"
+    # Managed by cisagov/skeleton-tf-module
+    ignore:
+      - dependency-name: "hashicorp/aws"
```
Comment (mcdonnnj) [inline .github/dependabot.yml]: No such directory in this project.

```suggestion
```

File: elasticsearchreadonly_role.tf
Comment (mcdonnnj) [inline elasticsearchreadonly_role.tf:1]: Would you please add `"sts:TagSession",` to the `elasticsearchreadonly_assume_role_doc` IAM policy document in addition to `"sts:AssumeRole",`.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
In addition to the suggestions would you please add `"sts:TagSession",` to https://github.com/cisagov/cool-dns-dmarc-import/blob/a5b936a514fd81b707a5283c6dd299259a6b3163/assume_role_policy_doc.tf#L8-L10.

File: elasticsearchreadonly_role.tf
Comment (dav3r) [reply elasticsearchreadonly_role.tf:1]: Done in https://github.com/cisagov/cool-dns-dmarc-import/pull/3/commits/7c061b3f0155b34631b13b1694f2f5241dd81a03 👍

File: .github/dependabot.yml (hunk 4 above)
Comment (mcdonnnj) [reply .github/dependabot.yml]: Sorry for missing the extra line x.x

**Review by jsf9k — COMMENTED** [review]
Oh for the love of JVPITER, how I so dearly wish I could approve this pull request!

File: setup-env [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    for parent_name, v in lineage["lineage"].items():
+        remote_url = v["remote-url"]
+        print(f"git remote add {parent_name} {remote_url};")
+        print(f"git remote set-url --push {parent_name} no_push;")
+else:
+    print(f'Unsupported lineage version: {lineage["version"]}', file=sys.stderr)
+END_OF_LINE
+)"
+
+# Qapla
```
Comment (jasonodoom) [inline setup-env:189]: It's missing the apostrophe.

```suggestion
# Qapla'
```

Comment (jsf9k) [reply setup-env:189]: That should be fixed via a separate PR in [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic) and allowed to trickle down during the next Kraken.

**Review by jasonodoom — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

File: provisiondmarcimport_policy.tf (hunk 3 above)
Comment (dav3r) [reply provisiondmarcimport_policy.tf]: Good call- I tested this and you are correct.  I made the changes in https://github.com/cisagov/cool-dns-dmarc-import/pull/3/commits/bdbc617df2cd6771607f320e14318e72c5d5d77e and applied them in Production.  I was able to verify that new log streams continued to flow into the lambda function log group and the count of documents in the dmarc reports DB continued to increase.

Comment (jsf9k) [reply provisiondmarcimport_policy.tf]: Thank you for doing that!

**Pull request merged**